code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +361,60 @@
 		P.fire()
 		return P
 
-/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
```

Comment:
this should not be removed.

Reply:
why not

Reply:
by why, i mean the same functionality is already provided by environment_smash
it doesnt even have a real purpose and is only called by destroysurroundings and gorilla when the environment_smash thing should probably be used instead
tell me  if im wrong because im new to the codebase, its just how i see it

Reply:
It lets you control what walls a mob can smash if it's able to smash walls, which is important.

Reply:
again, not trying to argue for arguments sake or anything, but its not really used right now and theres no real reason itll ever be used in the future (as far as i know)
i dont really see a difference in the types of walls we have so the difference that this proc talks about seems to be irrelevent

Reply:
also, environment_smash is never called aside from this proc and your suggested code doesnt even use it, so it seems superfluous to have a bunch of environment_smash variables scattered around the code and a cansmashturfs proc